### PR TITLE
Update pen/drawings to properly use environment maps

### PR DESCRIPTION
**Pull request by InfiniteLee**
set environment map for pens and drawings; create a three mesh for pen tip instead of using an a-sphere; use MobileStandardShader when app quality is low.

876941a0180d639a0bd5052764ff3a1c

### CODE REVIEW

File: src/components/tools/pen.js [hunk 1]
```diff
@@ -138,6 +139,22 @@ AFRAME.registerComponent("pen", {
...
+    this.penTip = new THREE.Mesh(new THREE.SphereGeometry(1, 16, 12), material);
+    this.penTip.scale.setScalar(this.data.radius / this.el.parentEl.object3D.scale.x);
+    this.penTip.matrixNeedsUpdate = true;
+
+    this.el.setObject3D("mesh", this.penTip);
+
+    const environmentMapComponent = this.el.sceneEl.components["environment-map"];
+    if (environmentMapComponent) {
+      environmentMapComponent.applyEnvironmentMap(this.el.object3D);
+      environmentMapComponent.applyEnvironmentMap(this.el.parentEl.object3D);
```
Comment (brianpeiris) [inline src/components/tools/pen.js]: What is `parentEl` here? `applyEnvironmentMap` will traverse children anyway, so perhaps you can avoid the first call to it?

Comment (InfiniteLee) [reply src/components/tools/pen.js]: the `parentEl` is the actual pen model- I didn't know that `applyEnvironmentMap` traverses so I can just call it on that.

File: src/components/tools/pen.js [hunk 2]
```diff
@@ -138,6 +139,22 @@ AFRAME.registerComponent("pen", {
     this.setDirty = this.setDirty.bind(this);
     this.dirty = true;
 
+    let material = new THREE.MeshStandardMaterial();
+    if (window.APP && window.APP.quality === "low") {
+      material = MobileStandardMaterial.fromStandardMaterial(material);
+    }
+    this.penTip = new THREE.Mesh(new THREE.SphereGeometry(1, 16, 12), material);
```
[the remark this answers is not in the document]
Comment (InfiniteLee) [reply src/components/tools/pen.js]: Yep will do, thanks.

File: src/components/tools/pen.js [hunk 3]
```diff
@@ -157,7 +174,7 @@ AFRAME.registerComponent("pen", {
 
   update(prevData) {
     if (prevData.color != this.data.color) {
-      this.el.setAttribute("color", this.data.color);
+      this.penTip.material.color.set(this.data.color);
```
[the remark this answers is not in the document]
Comment (InfiniteLee) [reply src/components/tools/pen.js]: I originally wanted to do that, however AFrame component schemas don't support hex values- the closest you can get is a "color" property which is stored as a string. We could just not use the schema properties for color in this case, but regardless it requires a bit of downstream changes in `networked-drawing` that I wouldn't want to change in this PR. I'll consider addressing this in a future update.

**Pull request merged**